zerver/lib/actions.py

```diff
@@ -2609,6 +2609,22 @@ def truncate_topic(topic):
     # type: (Text) -> Text
     return truncate_content(topic, MAX_SUBJECT_LENGTH, "...")
 
+def do_update_buddy_list(user_profile, buddy_profile, should_add):
+    # type: (UserProfile, str, bool) -> None
+
+    record = BuddyList.objects.get_or_create(user=user_profile, buddy=buddy_profile)
```

Comment:
We definitely don't want to create it for the delete case...it's possible that Django has some kind of delete-if-exists helper function in its API, but if not it's fine to first check if it exists, and if it does exist, only then delete it.